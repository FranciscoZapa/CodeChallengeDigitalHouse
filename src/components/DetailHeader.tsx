import {Box, Heading} from 'native-base';
import React, {PropsWithChildren} from 'react';
import {styles} from '../styles/Detail/DetailScreenStyles';

type DetailHeaderProps = PropsWithChildren<{
  product: string;
}>;

export const DetailHeader = ({product}: DetailHeaderProps) => {
  return (
    <Box height={'15%'} style={styles.title}>
      <Heading style={styles.title}>{product}</Heading>
    </Box>
  );
};
