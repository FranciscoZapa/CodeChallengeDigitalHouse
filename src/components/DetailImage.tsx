import {Box, Image} from 'native-base';
import React, {PropsWithChildren} from 'react';
import {styles} from '../styles/Detail/DetailImageStyles';

type DetailImageProps = PropsWithChildren<{image: string; product: string}>;

export const DetailImage = ({image, product}: DetailImageProps) => {
  return (
    <Box
      alignItems={'center'}
      justifyContent={'center'}
      height={'48%'}
      style={styles.box}>
      <Box
        style={styles.imageBorder}
        rounded={'lg'}
        height={'99%'}
        width={'95%'}>
        <Image
          rounded={'lg'}
          source={{
            uri: image,
          }}
          alt={product}
          height={'67%'}
          width={'63%'}
        />
      </Box>
    </Box>
  );
};
