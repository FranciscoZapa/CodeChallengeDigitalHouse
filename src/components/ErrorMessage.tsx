import React from 'react';
import {Box, HStack, Heading} from 'native-base';
import {errorlabel} from '../utils/constants';
import {styles} from '../styles/Status/ErrorMessageStyles';

export const ErrorMessage = () => (
  <Box
    alignItems="center"
    justifyContent="space-evenly"
    height={'100%'}
    style={styles.box}>
    <HStack space={2} justifyContent="center">
      <Heading color="white" fontSize="md">
        {errorlabel}
      </Heading>
    </HStack>
  </Box>
);
