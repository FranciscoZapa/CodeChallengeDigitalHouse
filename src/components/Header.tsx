import {Box, Text} from 'native-base';
import React from 'react';
import {styles} from '../styles/Header/HeaderStyles';
import {welcome, userName} from '../utils/constants';

export const Header = () => {
  return (
    <Box height={'10%'} style={styles.box}>
      <Text style={styles.heading}>{welcome}</Text>
      <Text style={styles.text}>{userName}</Text>
    </Box>
  );
};
