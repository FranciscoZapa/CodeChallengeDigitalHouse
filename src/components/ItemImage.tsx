import {Image} from 'native-base';
import React, {PropsWithChildren} from 'react';

type ImageProps = PropsWithChildren<{image: string; product: string}>;

export const ItemImage = ({image, product}: ImageProps) => {
  return (
    <Image
      rounded={'lg'}
      source={{
        uri: image,
      }}
      alt={product}
      size="sm"
    />
  );
};
