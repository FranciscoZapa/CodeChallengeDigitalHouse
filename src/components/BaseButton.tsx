import {Button, Text} from 'native-base';
import React, {PropsWithChildren} from 'react';
import {styles} from '../styles/List/Buttonstyles';

type BaseButtonProps = PropsWithChildren<{
  isFull: boolean;
  label: string;
  onPress: any;
}>;

export const BaseButton = ({isFull, label, onPress}: BaseButtonProps) => {
  return (
    <Button
      onPress={onPress}
      rounded={'lg'}
      size={'lg'}
      width={isFull ? '80%' : '40%'}
      style={styles.button}>
      <Text style={styles.buttonLabel}>{label}</Text>
    </Button>
  );
};
