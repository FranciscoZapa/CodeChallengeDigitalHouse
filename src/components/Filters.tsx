import {HStack} from 'native-base';
import React, {PropsWithChildren} from 'react';
import {alllabel, changedlabel, gainlabel} from '../utils/constants';
import {BaseButton} from './BaseButton';

type FiltersProps = PropsWithChildren<{
  filterActivated: boolean;
  handleAll: any;
  handleGain: any;
  handleChanged: any;
}>;

export const Filters = ({
  filterActivated,
  handleAll,
  handleGain,
  handleChanged,
}: FiltersProps) => (
  <HStack
    space={4}
    justifyContent={'center'}
    alignItems={'center'}
    height={'17%'}>
    {filterActivated ? (
      <BaseButton isFull={true} label={alllabel} onPress={handleAll} />
    ) : (
      <>
        <BaseButton isFull={false} label={gainlabel} onPress={handleGain} />
        <BaseButton
          isFull={false}
          label={changedlabel}
          onPress={handleChanged}
        />
      </>
    )}
  </HStack>
);
