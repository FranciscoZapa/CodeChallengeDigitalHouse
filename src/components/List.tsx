import {Box, FlatList, Text} from 'native-base';
import React, {PropsWithChildren, useMemo, useState} from 'react';
import {styles} from '../styles/List/ListStyles';
import {ListItemStruct} from '../types/List/ListTypes';
import {movements} from '../utils/constants';
import {Filters} from './Filters';
import {ListItem} from './ListItem';

type ListProps = PropsWithChildren<{
  list: Array<ListItemStruct> | null;
  setFilteredList: any;
  navigation: any;
}>;

export const List = ({list, setFilteredList, navigation}: ListProps) => {
  const [filter, setFilter] = useState('all');
  const filterActivated = useMemo(() => {
    if (filter === 'all') {
      return false;
    }
    return true;
  }, [filter]);
  const filteredList = useMemo(() => {
    if (list && list.length > 0) {
      if (filter === 'all') {
        setFilteredList(list);
        return list;
      }
      const filterIsRedemption = filter === 'changed' ? true : false;
      const filtered = list.filter(
        item => item.is_redemption === filterIsRedemption,
      );
      setFilteredList(filtered);
      return filtered;
    } else {
      return [];
    }
  }, [list, filter, setFilteredList]);
  const handleAll = () => {
    setFilter('all');
  };
  const handleGain = () => {
    setFilter('gain');
  };
  const handleChanged = () => {
    setFilter('changed');
  };

  return (
    <Box height={'65%'}>
      <Text style={styles.text}>{movements}</Text>
      <Box height={'76%'} style={styles.listContainer}>
        <FlatList
          style={styles.flatlist}
          data={filteredList}
          renderItem={({item}) => (
            <ListItem
              product={item.product}
              createdAt={item.createdAt}
              points={item.points}
              image={item.image}
              is_redemption={item.is_redemption}
              id={item.id}
              navigation={navigation}
            />
          )}
          keyExtractor={item => item.id}
        />
      </Box>
      <Filters
        filterActivated={filterActivated}
        handleAll={handleAll}
        handleGain={handleGain}
        handleChanged={handleChanged}
      />
    </Box>
  );
};
