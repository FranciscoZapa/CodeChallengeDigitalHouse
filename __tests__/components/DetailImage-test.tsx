import React from 'react';
import renderer from 'react-test-renderer';
import {NativeBaseProvider} from 'native-base';
import {DetailImage} from '../../src/components/DetailImage';

describe('DetailImage', () => {
  it('should render correctly', () => {
    const product = 'producto';
    const image = 'https://loremflickr.com/640/480/transport';
    const tree = renderer
      .create(
        <NativeBaseProvider>
          <DetailImage product={product} image={image} />
        </NativeBaseProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
