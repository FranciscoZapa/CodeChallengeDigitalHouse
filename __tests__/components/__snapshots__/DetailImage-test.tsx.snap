// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailImage should render correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
